src/app/icd/client/ICDClientInfoPersistentStorage.h

```diff
@@ -0,0 +1,95 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "ICDClientInfo.h"
+#include "ICDStorageKeyDelegate.h"
+#include <crypto/CHIPCryptoPAL.h>
+#include <lib/core/CHIPConfig.h>
+#include <lib/core/CHIPPersistentStorageDelegate.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/core/ScopedNodeId.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/CommonIterator.h>
+#include <lib/support/StorageKeyName.h>
+#include <stddef.h>
+
+namespace chip {
+namespace app {
+
+/**
+ * The ICDClientInfoPersistentStorage class is an abstract interface that defines the operations
+ * for storing, retrieving and deleting ICD client information in persistent storage.
+ */
+class ICDClientInfoPersistentStorage
```

Comment:
Suggest rename to "ICDClientStorage" --> shorter and reduces needless words. "Persistent" is relative, and it may not need to be persistently stored by itself if there is alternate storage to back an implementation.

Reply:
Done, thanks